[iOS] Make it possible to use WebKit1 without a web thread

cdumez opened the pull request:
#### 038f40cdb52d8dd888549503a56cca3a1c692d45
<pre>
[iOS] Make it possible to use WebKit1 without a web thread
<a href="https://bugs.webkit.org/show_bug.cgi?id=176812">https://bugs.webkit.org/show_bug.cgi?id=176812</a>

Reviewed by Ryosuke Niwa.

Make it possible to use WebKit1 without a web thread.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/WebCoreMainThread.cpp: Added.
(WebCore::shouldUseWebThread):
(WebCore::initializeMainThreadIfNeeded):
* Source/WebCore/platform/WebCoreMainThread.h: Added.
* Source/WebCore/platform/ios/wak/WebCoreThread.mm:
(WebThreadIsLocked):
* Source/WebKitLegacy/mac/History/WebBackForwardList.mm:
(+[WebBackForwardList initialize]):
* Source/WebKitLegacy/mac/History/WebHistoryItem.mm:
(+[WebHistoryItem initialize]):
* Source/WebKitLegacy/mac/Misc/WebCache.mm:
(+[WebCache initialize]):
* Source/WebKitLegacy/mac/Misc/WebElementDictionary.mm:
(+[WebElementDictionary initialize]):
* Source/WebKitLegacy/mac/Misc/WebIconDatabase.mm:
* Source/WebKitLegacy/mac/Misc/WebStringTruncator.mm:
(+[WebStringTruncator initialize]):
* Source/WebKitLegacy/mac/Plugins/WebBasePluginPackage.mm:
(+[WebBasePluginPackage initialize]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebEditorClient.mm:
(+[WebUndoStep initialize]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
(+[WebFramePolicyListener initialize]):
* Source/WebKitLegacy/mac/WebView/WebArchive.mm:
(+[WebArchivePrivate initialize]):
* Source/WebKitLegacy/mac/WebView/WebDataSource.mm:
(+[WebDataSource initialize]):
(-[WebDataSource _mainDocumentError]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(+[WebHTMLViewPrivate initialize]):
(+[WebHTMLView initialize]):
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(+[WebPreferences initialize]):
* Source/WebKitLegacy/mac/WebView/WebResource.mm:
(+[WebResourcePrivate initialize]):
* Source/WebKitLegacy/mac/WebView/WebTextIterator.mm:
(+[WebTextIteratorPrivate initialize]):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView enableWebThread]):
(+[WebView initialize]):
(-[WebView stopLoadingAndClear]):
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
(+[WebViewPrivate initialize]):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(dumpRenderTree):
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/JSLockTakesWebThreadLock.mm:
(TestWebKitAPI::TEST(WebKitLegacy, JSLockTakesWebThreadLock)):
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/WebGLNoCrashOnOtherThreadAccess.mm:
(TestWebKitAPI::TEST(WebKitLegacy, WebGLNoCrashOnOtherThreadAccess)):
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/WebGLPrepareDisplayOnWebThread.mm:
(TestWebKitAPI::TEST(WebKitLegacy, WebGLPrepareDisplayOnWebThread)):

Canonical link: <a href="https://commits.webkit.org/300662@main">https://commits.webkit.org/300662@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af05ae2e8343816ee251ae318133123a8fc12f93

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123374 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43089 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33785 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130088 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75498 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/39cb0b5b-dfaf-4fc8-9cf9-a4ca94876404) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125251 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43812 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51683 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93781 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62222 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3c6c69bf-36a7-483e-82a6-3d78400c77e2) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126327 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34908 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110382 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74409 "Found 2 new API test failures: TestWTF:WTF_Condition.OneProducerOneConsumerOneSlotTimeout, TestWTF:WTF_Condition.OneProducerOneConsumerOneSlot (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e0f8f5bc-8f5d-43fa-bfff-15197f7d60f0) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33875 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28540 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73605 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104622 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28766 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132805 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50324 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38299 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102272 "Found 1 new test failure: imported/blink/fast/pagination/viewport-y-horizontal-bt-rtl.html (failure)") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50700 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106605 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102125 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25967 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47479 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25703 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/47114 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50179 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55940 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49650 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53000 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51328 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/platform/WebCoreMainThread.cpp:
```diff
@@ -0,0 +1,56 @@
+
+bool shouldUseWebThread()
+{
+#if PLATFORM(IOS_FAMILY) && !ENABLE(WEB_THREAD_DISABLEMENT)
```
I don't see `WEB_THREAD_DISABLEMENT` being defined anywhere??

cdumez replied:
I will follow-up to define it, right now this PR does not change behavior. Please see my comments on Slack for the reasoning.

cdumez committed:
[iOS] Make it possible to use WebKit1 without a web thread
https://bugs.webkit.org/show_bug.cgi?id=176812

Reviewed by Ryosuke Niwa.

Make it possible to use WebKit1 without a web thread.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/WebCoreMainThread.cpp: Added.
(WebCore::shouldUseWebThread):
(WebCore::initializeMainThreadIfNeeded):
* Source/WebCore/platform/WebCoreMainThread.h: Added.
* Source/WebCore/platform/ios/wak/WebCoreThread.mm:
(WebThreadIsLocked):
* Source/WebKitLegacy/mac/History/WebBackForwardList.mm:
(+[WebBackForwardList initialize]):
* Source/WebKitLegacy/mac/History/WebHistoryItem.mm:
(+[WebHistoryItem initialize]):
* Source/WebKitLegacy/mac/Misc/WebCache.mm:
(+[WebCache initialize]):
* Source/WebKitLegacy/mac/Misc/WebElementDictionary.mm:
(+[WebElementDictionary initialize]):
* Source/WebKitLegacy/mac/Misc/WebIconDatabase.mm:
* Source/WebKitLegacy/mac/Misc/WebStringTruncator.mm:
(+[WebStringTruncator initialize]):
* Source/WebKitLegacy/mac/Plugins/WebBasePluginPackage.mm:
(+[WebBasePluginPackage initialize]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebEditorClient.mm:
(+[WebUndoStep initialize]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
(+[WebFramePolicyListener initialize]):
* Source/WebKitLegacy/mac/WebView/WebArchive.mm:
(+[WebArchivePrivate initialize]):
* Source/WebKitLegacy/mac/WebView/WebDataSource.mm:
(+[WebDataSource initialize]):
(-[WebDataSource _mainDocumentError]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(+[WebHTMLViewPrivate initialize]):
(+[WebHTMLView initialize]):
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(+[WebPreferences initialize]):
* Source/WebKitLegacy/mac/WebView/WebResource.mm:
(+[WebResourcePrivate initialize]):
* Source/WebKitLegacy/mac/WebView/WebTextIterator.mm:
(+[WebTextIteratorPrivate initialize]):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView enableWebThread]):
(+[WebView initialize]):
(-[WebView stopLoadingAndClear]):
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
(+[WebViewPrivate initialize]):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(dumpRenderTree):
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/JSLockTakesWebThreadLock.mm:
(TestWebKitAPI::TEST(WebKitLegacy, JSLockTakesWebThreadLock)):
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/WebGLNoCrashOnOtherThreadAccess.mm:
(TestWebKitAPI::TEST(WebKitLegacy, WebGLNoCrashOnOtherThreadAccess)):
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/WebGLPrepareDisplayOnWebThread.mm:
(TestWebKitAPI::TEST(WebKitLegacy, WebGLPrepareDisplayOnWebThread)):

Canonical link: https://commits.webkit.org/300662@main
